File: Python/Recursion/sum.py
# Function to add numbers till n
def Sum(n):
    sum = 0
    if n == 0 or n == 1:
        return n;
    else:
        return sum + Sum(n - 1)

# Main starts here
n = int(input("Enter the number n: "))
print("The sum is: ", Sum(n))

File: C++/Recursion/sum.cpp
#include <iostream>
using namespace std;

int Sum(int n) {
  int sum = n;
  if(n == 0 || n == 1) {
    return n;
  }
  else {
    return sum + Sum(n - 1);
  }
}

int main() {
  int n;
  cout << "Enter the number: ";
  cin >> n;
  cout << "The sum uptil number " << n << " is: " << Sum(n) << endl;
  return 0; 
}
